File: stlm-java/src/sequence/Text.py
#!/usr/bin/env python
""" generated source for module Text """
# package: sequence
import java.util.HashMap

import java.util.LinkedList

import java.util.Set

class Text(object):
    """ generated source for class Text """
    # 
    # 	 * Holds the "vocabulary" for the tree
    # 	 
    items = LinkedList()
    vocab = HashMap()
    index = int()

    def __init__(self):
        """ generated source for method __init__ """
        self.vocab = HashMap()
        self.items = LinkedList()
        self.index = 0
        # vocab.put((T)Constants.ROOT, index++);

    def getVocabulary(self):
        """ generated source for method getVocabulary """
        return self.vocab.keySet()

    def pushFront(self, item):
        """ generated source for method pushFront """
        self.items.addFirst(item)

    def pushBack(self, item):
        """ generated source for method pushBack """
        self.items.add(item)

    def put(self, word, i):
        """ generated source for method put """
        self.vocab.put(word, i)
        self.index += 1

    def pop(self):
        """ generated source for method pop """
        return self.items.pop()

    def cut(self):
        """ generated source for method cut """
        self.items.removeFirst()

    def size(self):
        """ generated source for method size """
        return len(self.items)

    def clear(self):
        """ generated source for method clear """
        self.items.clear()

    def at(self, i):
        """ generated source for method at """
        if len(self) == 0:
            return 0
        if i == -1:
            i = len(self) - 1
        return self.items.get(i)

    def print_(self):
        """ generated source for method print_ """

    def offer(self, word):
        """ generated source for method offer """
        if get(word) == -1:
            self.vocab.put(word, self.index)
            return self.index += 1
        return self.vocab.get(word)

    def get(self, word):
        """ generated source for method get """
        if self.vocab.containsKey(word):
            return self.vocab.get(word)
        return -1

    def has(self, word):
        """ generated source for method has """
        if self.get(word) == -1:
            return False
        return True

    def vocabSize(self):
        """ generated source for method vocabSize """
        return len(self.vocab)

    def getWordFromVocabIndex(self, ind):
        """ generated source for method getWordFromVocabIndex """
        for word in vocab.keySet():
            if self.vocab.get(word) == ind:
                return word
        # return (T)Constants.UNK;
        return None

    def getWordFromIndex(self, ind):
        """ generated source for method getWordFromIndex """
        for word in vocab.keySet():
            if self.vocab.get(word) == self.at(ind):
                return word
        # return (T)Constants.UNK;
        return None

File: stlm-java/src/sequence/Text.java
package sequence;

import java.util.HashMap;
import java.util.LinkedList;
import java.util.Set;

public class Text<T> {
	
	/*
	 * Holds the "vocabulary" for the tree
	 */

	public LinkedList<Integer> items;
	HashMap<T,Integer> vocab;
    int index;

    
    public Text() {
    	vocab = new HashMap<T,Integer>();
    	items = new LinkedList<Integer>();
    	this.index = 0;
    	//vocab.put((T)Constants.ROOT, index++);
    }
    
    public Set<T> getVocabulary() {
    	return vocab.keySet();
    }
    
    public void pushFront(int item) {
    	items.addFirst(item);
    }
    
    public void pushBack(int item) {
    	items.add(item);
    }
    
    public void put(T word, int i) {
    	vocab.put(word, i);
    	index++;
    }
    
    public int pop() {
    	return items.pop();
    }
    
    public void cut() {
    	items.removeFirst();
    }
    
    public int size() {
    	return this.items.size();
    }
    
    public void clear() {
    	this.items.clear();
    }
    
    public int at(int i) {
    	if (size() == 0) return 0;
    	if (i == -1) i = size() -1;
    	return items.get(i);
    }
    
    public void print() {
    	
    }
    
    public int offer(T word) {
    	if (get(word) == -1){
    		vocab.put(word, index);
    		return index++;
    	}
    	return vocab.get(word);
    }
    
    public int get(T word) {
    	if (vocab.containsKey(word)) {
    		return vocab.get(word);
    	}
    	return -1;
    }
    
    public boolean has(T word) {
    	if (get(word) == -1) 
    		return false;
    	return true;
    }
    
    public int vocabSize() {
    	return this.vocab.size();
    }
    
    public T getWordFromVocabIndex(int ind) {
    	for (T word : vocab.keySet()) {
    		if (vocab.get(word) == ind)
    			return word;
    	}
    	//return (T)Constants.UNK;
    	return null;
    }
    
    public T getWordFromIndex(int ind) {

    	for (T word : vocab.keySet()) {
    		if (vocab.get(word) == this.at(ind))
    			return word;
    	}
    	//return (T)Constants.UNK;
    	return null;
    }    
    

}
